### PR TITLE
Address a number of smaller vscode issues

### LINKED ISSUE #59: VSCode ext: `ccdrec` snippet uses `mut` host without `mutable` attribute
**Bug Description**

In the VSCode extension, the `ccdrec` snippet uses `mut` host without `mutable` attribute, which causes a compiler error.

**Steps to Reproduce**
1. Use the snippet `ccdrec` and enter values of existing data types and contracts in the placeholders.
2. Return a value from the function.
3. Compile

**Expected Result**
Compiles successfully.

**Actual Result**
A compiler error because the snippet uses `&mut impl HasHost...` instead of `&impl HasHost` when the `mutable` attribute isn't specified.
```
error[E0308]: mismatched types
   --> src/lib.rs:105:1
    |
105 | / #[receive(
106 | |     contract = "vstest",
107 | |     name = "foo",
108 | |     parameter = "()",
109 | |     return_value = "()",
110 | |     error = "()"
111 | | )]
    | |__^ types differ in mutability
112 |   fn vstest_foo<S: HasStateApi>(
    |      ---------- arguments to this function are incorrect
    |
    = note: expected mutable reference `&mut _`
                       found reference `&ExternHost<_>`
note: function defined here
   --> src/lib.rs:112:4
    |
112 | fn vstest_foo<S: HasStateApi>(
    |    ^^^^^^^^^^
113 |     ctx: &impl HasReceiveContext,
114 |     host: &mut impl HasHost<State, StateApiType = S>
    |     ------------------------------------------------
    = note: this error originates in the attribute macro `receive` (in Nightly builds, run with -Z macro-backtrace for more info)
```

**Versions**
- Software Version: 0.1.0-0
- OS: macOS Ventura (Apple M2 chip)

### LINKED ISSUE #60: VSCode ext: Small issues in readme
Found two small issues in the readme of the VSCode extension.
1.  `all prefixed with ccd (for smart contract)` should be `all prefixed with ccd (for Concordium)`.
2. `The extension relies on [rustup](https://rustup.rs/) and cargo being installed and available in PATH.` It links to rustup but not cargo. I think it should link to both.

### LINKED ISSUE #61: VSCode ext: Build task does not use `--schema-embed`
**Bug Description**

The configurable build task does not use `--schema-embed`, but the default build task does.
I would expect that it matches exactly by default, and then you can configure it to your liking.

**Steps to Reproduce**
1. Run `Tasks: Configure Task`
2. Choose `.: Build smart contract`
3. You are directed to the file `.vscode/tasks.json`

**Expected Result**
The args for concordium build should include `--schema-embed`.

**Actual Result**
`--schema-embed` is missing:
```
		{
			"type": "concordium",
			"command": "build",
			"cwd": ".",
			"args": [
				"--out",
				"concordium-out/module.wasm.v1",
				"--schema-json-out",
				"concordium-out",
				"--schema-base64-out",
				"concordium-out/module-schema.bs64"
			],
			"problemMatcher": [],
			"label": ".: Build smart contract"
		},
```

**Versions**
- Software Version: 0.1.0-0
- OS: macOS Ventura (Apple M2)

### LINKED ISSUE #62: VSCode ext: Custom test task does not run tests, only `cargo build`
**Bug Description**

The configurable test task does not run the tests, but it does seem to run `cargo build` without any arguments.

**Steps to Reproduce**
1. Run `Tasks: Configure Task`
2. Choose `.: Test smart contract`
3. The file `.vscode/tasks.json` is created.
    With content:
   ```
        ...
	{
			"type": "concordium",
			"command": "test",
			"cwd": ".",
			"args": [],
			"problemMatcher": [],
			"label": ".: Test smart contract"
		}
         ...
   ```
4. Run the task with `Tasks: Run Task` 
5. Then choose `.: Test smart contract`

**Expected Result**
In step 3 I would assume that it should match the exact behavior and args of the default concordium test command. 
In step 5, I would expect the tests to run.

**Actual Result**
In step 3: it doesn't match the args etc that I would expect.
In step 5: It only compiles, and no tests are run:
```
*  Executing task: /Users/kasper/.vscode/extensions/concordium.concordium-smart-contracts-1.0.0/executables/cargo-concordium concordium build 

   Compiling vstest v0.1.0 (/Users/kasper/Developer/Concordium/vstest/vstest)
    Finished release [optimized] target(s) in 0.11s
    Finished smart contract module 21.515 kB
 *  Terminal will be reused by tasks, press any key to close it. 
```

**Versions**
- Software Version: 0.1.0-0
- OS: macOS Ventura (Apple M2)

### LINKED ISSUE #63: VSCode ext: Custom tasks prefixed with `.:` instead of `ccd:` or `concordium:`
**Bug Description**

The configurable tasks are prefixed with `.:` instead of `ccd:` or `concordium:`.
In comparison, the regular rust tasks are prefixed with `rust:`.
I assume that this can lead to confusion, if you also have tasks for smart contracts on other chains. For example for the build task, which has the full name `.: Build smart contract`

**Steps to Reproduce**
1. Run `Tasks: Configure Task`
2. See the names of the different tasks available

**Expected Result**
`rust: cargo clean`
...
`concordium: Build smart contract`
`concordium: Test smart contract`

**Actual Result**
`rust: cargo clean`
...
`.: Build smart contract`
`.: Test smart contract`

**Versions**
- Software Version: 0.1.0-0
- OS: macOS Ventura (Apple M2)

### PR DESCRIPTION
**Pull request by limemloh**
## Purpose

Fixes #59
Fixes #60
Fixes #61
Fixes #62
Fixes #63

### CODE REVIEW

File: vscode-smart-contracts/src/extension.ts [hunk 1]
```diff
@@ -125,7 +126,7 @@ const taskProvider: vscode.TaskProvider = {
       const workspaceFolder = vscode.workspace.getWorkspaceFolder(
         vscode.Uri.file(cwd)
       );
-      const resolvedTask = await cargoConcordium.build(
+      const resolvedTask = await cargoConcordium.test(
```
Comment (Bargsteen) [inline vscode-smart-contracts/src/extension.ts:129]: Will this run the tests using `cargo concordium test`? Or just `cargo test`?

**Review by Bargsteen — COMMENTED** [review]
Looks good, but I am a bit unsure about how the test task works now.

File: vscode-smart-contracts/src/extension.ts (hunk 1 above)
Comment (limemloh) [reply vscode-smart-contracts/src/extension.ts:129]: It will use `cargo concordium test`
The extension does not provide anything for running `cargo test`

Comment (Bargsteen) [reply vscode-smart-contracts/src/extension.ts:129]: Ok, great!

**Review by Bargsteen — APPROVED** [review]

**Pull request merged**